src/gui/widgets/XournalWidget.h

```diff
@@ -29,63 +23,46 @@ class XournalView;
 class InputContext;
 class NewGtkInputDevice;
 
+#include <memory>
 
-typedef struct _GtkXournal GtkXournal;
-typedef struct _GtkXournalClass GtkXournalClass;
-
-struct _GtkXournal {
-    GtkWidget widget;
-
-    /**
-     * The view class
-     */
-    XournalView* view;
+class XournalWidget {
+public:
+    XournalWidget(XournalView* view, std::unique_ptr<InputContext> inputContext);
+    XournalWidget(XournalView* view, ScrollHandling* scrollHandling);
+    virtual ~XournalWidget();
 
-    /**
-     * Scrollbars
-     */
-    ScrollHandling* scrollHandling;
+    /** Current selection */
+    EditSelection* selection;
```

Comment:
This is undefined behaviour, since selection is used after initialisation.
Todo: Either use smart pointers or initialize the variable with `EditSelection* selection{}`;

Reply:
Good catch, I missed this when refactoring. I will probably get around to updating the PR this weekend.

Reply:
The best thing we can do right now (without refactoring literally everything else) is to initialize `selection` to null, since `XournalView::setSelection` passes in a raw pointer and assumes that the widget will take ownership.